src/Raven.Server/Documents/Replication/Incoming/AbstractIncomingReplicationHandler.cs

```diff
@@ -489,8 +492,7 @@ protected void ReceiveSingleDocumentsBatch(TOperationContext context, int replic
                                 writer.Flush();
                             }
 
-                            HandleTaskCompleteIfNeeded();
-                            task = null;
+                            HandleMissingAttachmentsIfNeeded();
```

Comment:
I think it would be better to remove this and handle this all in `HandleBatchAsync`

Reply:
Done